atomate/vasp/firetasks/exchange_tasks.py

```diff
@@ -0,0 +1,458 @@
+        formula = self["parent_structure"].formula
+        formula_pretty = self["parent_structure"].composition.reduced_formula
+
+        # Get magnetic orderings collection from db
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+        mmdb.collection = mmdb.db["magnetic_orderings"]
+
+        # Look up static calcs if needed
+        if not self["avg"]:
+
+            # Get documents
+            docs = list(
+                mmdb.collection.find(
+                    {"wf_meta.wf_uuid": wf_uuid},
+                    ["task_id", "structure", "energy_per_atom"],
+                )
+            )
+
+            # Get structures and energy / unit cell
+            structures = [Structure.from_dict(d["structure"]) for d in docs]
+            epas = [d["energy_per_atom"] for d in docs]
+            n_atoms = len(structures[0])
+            energies = [e * n_atoms for e in epas]
+        else:
+            structures = self["structures"]
+            energies = self["energies"]
+
+        # Total energies
+        energies = [e * len(s) for e, s in zip(energies, structures)]
+
+        # Map system to a Heisenberg Model
+        hmapper = HeisenbergMapper(structures, energies, self["cutoff"], self["tol"])
+
+        # Get MSONable Heisenberg Model
+        hmodel = hmapper.get_heisenberg_model()
+        hmodel_dict = hmodel.as_dict()
+        name = "heisenberg_model_" + str(self["cutoff"])
+
+        wf_meta = {"wf_uuid": wf_uuid}
+        task_doc = {
+            "wf_meta": wf_meta,
+            "formula_pretty": formula_pretty,
+            "nn_cutoff": self["cutoff"],
+            "nn_tol": self["tol"],
+            "heisenberg_model": hmodel_dict,
+        }
+
+        if fw_spec.get("tags", None):
+            task_doc["tags"] = fw_spec["tags"]
+
+        # Exchange collection
+        mmdb.collection = mmdb.db["exchange"]
+        mmdb.collection.insert(task_doc)
+
+
+@explicit_serialize
+class HeisenbergConvergence(FiretaskBase):
+    """
+    Quick check to see if the Heisenberg model has "converged" for any particular nearest neighbor cutoff value in the sense that |J_ij| < |E_0| for all i, j. 
+
+    If not, it doesn't make sense to do Monte Carlo and the system is either 1) not well described by the Heisenberg Model (perhaps the magnetic moments are not localized) or 2) not correctly captured by the provided low-energy magnetic orderings.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        exchange_wf_uuid (int): Unique id for record keeping.
+        parent_structure (Structure): Structure object.
+        avg (bool): <J> exchange param only.
+
+    TODO:
+        * More robust convergence check
+
+    """
+
+    required_params = ["db_file", "exchange_wf_uuid", "parent_structure", "avg"]
+    optional_params = []
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["exchange_wf_uuid"]
```

Comment:
In keeping with the magnetism workflow, can you rename this parameter in all of your firetasks to just wf_uuid...